src/treemodel/operation/operation.js

```diff
@@ -81,11 +81,36 @@ export default class Operation {
 	 * @returns {Object} Clone of this object with the delta property replaced with string.
 	 */
 	toJSON() {
-		const json = clone( this );
+		const json = clone( this, true );
+
+		json.__class = this.constructor.className;
 
 		// Due to circular references we need to remove parent reference.
-		json.delta = this.delta ? '[engine.treeModel.Delta]' : null;
+		json.delta = this.delta ? `[${this.delta.constructor.className}]` : null;
 
 		return json;
 	}
+
+	/**
+	 * Name of the operation class used for serialization.
+	 *
+	 * @type {String}
+	 */
+	static get className() {
+		return 'engine.treeModel.operation.Operation';
+	}
+
+	/*jshint unused: true*/
```

Comment:
Base class doesn't need document - but all operations need document parameter.

Without `jshint unused: true` linter will throw errors.

This is to make methods consistent in documentation so all (base and child classes) heve the same signature.